Flaky `TestStringShannonEntropy` test

The [`TestStringShannonEntropy`](https://github.com/trufflesecurity/trufflehog/blob/def734a783b2d4542f2b353d326130d618f3c421/pkg/detectors/falsepositives_test.go#L153-L157) test fails periodically due to precision issues. 

This level of precision is unnecessary, and the tests should probably round to ~2 digits of precision.

```
 --- FAIL: TestStringShannonEntropy (0.00s)
    --- FAIL: TestStringShannonEntropy/entropy_2 (0.00s)
        falsepositives_test.go:177: StringShannonEntropy() = 0.22228483068568813, want 0.22228483068568816
    --- FAIL: TestStringShannonEntropy/entropy_3 (0.00s)
        falsepositives_test.go:177: StringShannonEntropy() = 0.22228483068568813, want 0.22228483068568816
FAIL
```

--- a/pkg/detectors/falsepositives_test.go
+++ b/pkg/detectors/falsepositives_test.go
@@ -3,7 +3,6 @@ package detectors
 import (
 	"context"
 	_ "embed"
-	"math"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -174,9 +173,11 @@ func TestStringShannonEntropy(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := roundToTwoDigits(StringShannonEntropy(tt.args.input))
-			if got != tt.want {
-				t.Errorf("StringShannonEntropy() = %v, want %v", got, tt.want)
+			got := StringShannonEntropy(tt.args.input)
+			if len(tt.args.input) > 0 && tt.want != 0 {
+				assert.InEpsilon(t, tt.want, got, 0.1)
+			} else {
+				assert.Equal(t, tt.want, got)
 			}
 		})
 	}
@@ -187,9 +188,4 @@ func BenchmarkDefaultIsKnownFalsePositive(b *testing.B) {
 		// Use a string that won't be found in any dictionary for the worst case check.
 		IsKnownFalsePositive("aoeuaoeuaoeuaoeuaoeuaoeu", DefaultFalsePositives, true)
 	}
-}
-
-func roundToTwoDigits(number float64) float64 {
-	scale := math.Pow(10, float64(2))
-	return math.Round(number*scale) / scale
 }

--- a/pkg/detectors/falsepositives_test.go
+++ b/pkg/detectors/falsepositives_test.go
@@ -3,6 +3,7 @@ package detectors
 import (
 	"context"
 	_ "embed"
+	"math"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -154,14 +155,14 @@ func TestStringShannonEntropy(t *testing.T) {
 			args: args{
 				input: "aaaaaaaaaaaaaaaaaaaaaaaaaaab",
 			},
-			want: 0.22228483068568816,
+			want: 0.22,
 		},
 		{
 			name: "entropy 3",
 			args: args{
 				input: "aaaaaaaaaaaaaaaaaaaaaaaaaaabaaaaaaaaaaaaaaaaaaaaaaaaaaab",
 			},
-			want: 0.22228483068568816,
+			want: 0.22,
 		},
 		{
 			name: "empty",
@@ -173,7 +174,8 @@ func TestStringShannonEntropy(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := StringShannonEntropy(tt.args.input); got != tt.want {
+			got := roundToTwoDigits(StringShannonEntropy(tt.args.input))
+			if got != tt.want {
 				t.Errorf("StringShannonEntropy() = %v, want %v", got, tt.want)
 			}
 		})
@@ -185,4 +187,9 @@ func BenchmarkDefaultIsKnownFalsePositive(b *testing.B) {
 		// Use a string that won't be found in any dictionary for the worst case check.
 		IsKnownFalsePositive("aoeuaoeuaoeuaoeuaoeuaoeu", DefaultFalsePositives, true)
 	}
+}
+
+func roundToTwoDigits(number float64) float64 {
+	scale := math.Pow(10, float64(2))
+	return math.Round(number*scale) / scale
 }

